client/app/reader/actions.js

```
@@ -43,34 +47,148 @@ export const onScrollToComment = (scrollToComment) => ({
   payload: { scrollToComment }
 });
 
-export const handlePlaceComment = () => ({
-  type: Constants.SET_COMMENT_FLOW_STATE,
+export const startEditAnnotation = (annotationId) => ({
+  type: Constants.START_EDIT_ANNOTATION,
   payload: {
-    state: Constants.PLACING_COMMENT_STATE
+    annotationId
   }
 });
 
-export const handleWriteComment = () => ({
-  type: Constants.SET_COMMENT_FLOW_STATE,
+export const openAnnotationDeleteModal = (annotationId) => ({
+  type: Constants.OPEN_ANNOTATION_DELETE_MODAL,
+  payload: {
+    annotationId
+  }
+});
+export const closeAnnotationDeleteModal = () => ({ type: Constants.CLOSE_ANNOTATION_DELETE_MODAL });
+export const selectAnnotation = (annotationId) => ({
+  type: Constants.SELECT_ANNOTATION,
   payload: {
-    state: Constants.WRITING_COMMENT_STATE
+    annotationId
   }
 });
 
-export const handleClearCommentState = () => ({
-  type: Constants.SET_COMMENT_FLOW_STATE,
+export const deleteAnnotation = (docId, annotationId) =>
+  (dispatch) => {
+    ApiUtil.delete(`/document/${docId}/annotation/${annotationId}`).
+      // I don't know why, but the request was not going through if
+      // we are not listening for the promise.
+      then(_.noop);
```

The `REQUEST_` methods update the redux store. API request failure is not handled yet; that's actually the whole point of doing this refactor. That was the AC for #1854, and we wanted to convert to redux before taking that on.

So, I may be sending you another PR for that later this week! :smile: